src/slot_provider/mined_blocks.rs

```diff
@@ -27,9 +27,12 @@ impl MinedBlocksSlotProvider {
 }
 
 impl SlotProvider for MinedBlocksSlotProvider {
-    fn run_for_every_slot<F>(&self, f: F) -> Box<dyn Future<Output = Result<()>> + Unpin + '_>
+    fn run_for_every_slot<'a, F>(
```

Comment:
Bit scared about these types but I wouldn't know how to do this better 😅 . I'm sure I'll try at some point.